azurerm/internal/services/media/media_streaming_endpoint_resource.go

```diff
@@ -0,0 +1,591 @@
+			Allow: &ipRanges,
+		}
+	}
+	// Get Akamai information
+	if raw, ok := accessControl["akamai_signature_header_authentication_key"]; ok {
+		akamaiSignatureKeyList := raw.([]interface{})
+		akamaiSignatureHeaderAuthenticationKeyList := make([]media.AkamaiSignatureHeaderAuthenticationKey, len(akamaiSignatureKeyList))
+		for index, akamaiSignatureKey := range akamaiSignatureKeyList {
+			akamaiKey := akamaiSignatureKey.(map[string]interface{})
+			base64Key := akamaiKey["base64_key"].(string)
+			expirationRaw := akamaiKey["expiration"].(string)
+			identifier := akamaiKey["identifier"].(string)
+
+			akamaiSignatureHeaderAuthenticationKey := media.AkamaiSignatureHeaderAuthenticationKey{
+				Base64Key:  utils.String(base64Key),
+				Identifier: utils.String(identifier),
+			}
+			if expirationRaw != "" {
+				expiration, _ := date.ParseTime(time.RFC3339, expirationRaw)
+				akamaiSignatureHeaderAuthenticationKey.Expiration = &date.Time{
+					Time: expiration,
+				}
+			}
+			akamaiSignatureHeaderAuthenticationKeyList[index] = akamaiSignatureHeaderAuthenticationKey
+			accessControlResult.Akamai = &media.AkamaiAccessControl{
+				AkamaiSignatureHeaderAuthenticationKeyList: &akamaiSignatureHeaderAuthenticationKeyList,
+			}
+		}
+	}
+
+	return accessControlResult
+}
+
+func flattenAccessControl(accessControl *media.StreamingEndpointAccessControl) []interface{} {
+	if accessControl == nil {
+		return make([]interface{}, 0)
+	}
+
+	result := make(map[string]interface{})
+
+	if accessControl.IP != nil {
+		allows := make([]interface{}, len(*accessControl.IP.Allow))
+		for i, ipAllow := range *accessControl.IP.Allow {
+			allow := make(map[string]interface{})
+			if ipAllow.Name != nil {
+				allow["name"] = ipAllow.Name
+			}
+			if ipAllow.Address != nil {
+				allow["address"] = ipAllow.Address
+			}
+			if ipAllow.SubnetPrefixLength != nil {
+				allow["subnet_prefix_length"] = ipAllow.SubnetPrefixLength
+			}
+			if allow != nil {
+				allows[i] = allow
+			}
+		}
+
+		result["ip_allow"] = allows
+	}
+
+	if accessControl.Akamai != nil {
+		akamaiSignatureKeyList := make([]interface{}, len(*accessControl.Akamai.AkamaiSignatureHeaderAuthenticationKeyList))
+		for i, key := range *accessControl.Akamai.AkamaiSignatureHeaderAuthenticationKeyList {
+			akamaiSignatureHeaderKey := make(map[string]interface{})
+			if key.Base64Key != nil {
+				akamaiSignatureHeaderKey["base64_key"] = key.Base64Key
+			}
+			if key.Expiration != nil {
+				akamaiSignatureHeaderKey["expiration"] = key.Expiration.Format(time.RFC3339)
+			}
+			if key.Identifier != nil {
+				akamaiSignatureHeaderKey["identifier"] = key.Identifier
+			}
+			if akamaiSignatureKeyList != nil {
+				akamaiSignatureKeyList[i] = akamaiSignatureHeaderKey
+			}
+		}
+
+		result["akamai_signature_header_authentication_key"] = akamaiSignatureKeyList
+	}
+
+	return []interface{}{result}
+}
+
+func expandCrossSiteAccessPolicies(d *schema.ResourceData) *media.CrossSiteAccessPolicies {
```

Comment:
rather than passing in `d` here - could we pass in the access policies:

```suggestion
func expandCrossSiteAccessPolicies(input []interface{}) *media.CrossSiteAccessPolicies {
```